colander/tests/test_colander.py

```diff
@@ -2787,6 +2787,32 @@ def test_clone(self):
         self.assertEqual(inner_clone.name, 'inner')
         self.assertEqual(inner_clone.foo, 2)
 
+    def test_clone_with_modified_schema_instance(self):
+        import colander
+        class Schema(colander.MappingSchema):
+            n1 = colander.SchemaNode(colander.String())
+            n2 = colander.SchemaNode(colander.String())
+        def compare_children(schema, cloned):
+            # children of the clone must match the cloned node's children and
+            # have to be clones themselves.
```

Comment:
Because [`zip` stops grouping at the shortest list](https://docs.python.org/3/library/functions.html#zip), we probably need to assert that the two lists are the same length first.